crates/miden-testing/src/kernel_tests/tx/test_active_note.rs

```diff
@@ -0,0 +1,494 @@
+use alloc::string::String;
+
+use anyhow::Context;
+use miden_lib::errors::tx_kernel_errors::ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_METADATA_WHILE_NO_NOTE_BEING_PROCESSED;
+use miden_lib::testing::mock_account::MockAccountExt;
+use miden_lib::utils::ScriptBuilder;
+use miden_objects::account::Account;
+use miden_objects::asset::FungibleAsset;
+use miden_objects::crypto::rand::{FeltRng, RpoRandomCoin};
+use miden_objects::note::{
+    Note,
+    NoteAssets,
+    NoteExecutionHint,
+    NoteInputs,
+    NoteMetadata,
+    NoteRecipient,
+    NoteTag,
+    NoteType,
+};
+use miden_objects::testing::account_id::{
+    ACCOUNT_ID_REGULAR_PRIVATE_ACCOUNT_UPDATABLE_CODE,
+    ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE,
+    ACCOUNT_ID_SENDER,
+};
+use miden_objects::{EMPTY_WORD, Felt, ONE, WORD_SIZE, Word};
+
+use crate::utils::create_p2any_note;
+use crate::{
+    Auth,
+    MockChain,
+    TransactionContextBuilder,
+    TxContextInput,
+    assert_transaction_executor_error,
+};
+
+#[test]
+fn test_get_sender_fails_from_tx_script() -> anyhow::Result<()> {
```

Comment:
Since these are "active note" tests, should we reflect that in the test name, e.g. `active_note_get_sender_fails_from_tx_script`? I guess this is quite a subjective thing though, so feel free to ignore. Also, we should probably discuss the "active note" name first before doing this to avoid unnecessary work.